sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsErrorCode.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// </summary>
+    public readonly partial struct TextAnalyticsErrorCode : IEquatable<TextAnalyticsErrorCode>
+    {
+        private readonly string _value;
+
+        /// <summary>
+        /// </summary>
+        /// <param name="value"></param>
+        public TextAnalyticsErrorCode(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        // Top-level codes
+        private const string InvalidRequestValue = "invalidRequest";
```

Comment:
Can we have these as enums instead?

Reply:
This is an ["extensible enum" type ](https://azure.github.io/azure-sdk/dotnet_implementation.html#dotnet-enums) we use when the service could potentially add values to the enum.  That would be a breaking change in .NET.